mvlearn/cluster/mv_coreg_spectral.py

```diff
@@ -0,0 +1,237 @@
+
+        Returns
+        -------
+        predictions : array-like, shape (n_samples,)
+            The predicted cluster labels for each sample.
+        '''
+
+        check_u_mats = list()
+
+        # Perform checks on inputted parameters and data
+        Xs = self._param_checks(Xs)
+        if self.v_lambda <= 0:
+            msg = 'v_lambda must be a positive value'
+            raise ValueError(msg)
+
+        # Compute the similarity matrices
+        sims = [self._affinity_mat(X) for X in Xs]
+
+        # Initialize matrices of eigenvectors
+        U_mats = []
+        L_mats = []
+        obj_vals = np.zeros((self._n_views, self.max_iter))
+        for ind in range(len(sims)):
+            u_mat, l_mat, o_val = self._init_umat(sims[ind])
+            U_mats.append(u_mat)
+            L_mats.append(l_mat)
+            obj_vals[ind, 0] = o_val
+
+        check_u_mats.append(U_mats[0])
+
+        # Iteratively solve for all U's
+        n_items = Xs[0].shape[0]
+        for it in range(1, self.max_iter):
+            # Performing alternating maximization by cycling through all
+            # pairs of views and updating all except view 1
+            for v1 in range(1, self._n_views):
+                # Computing the regularization term for view v1
+                l_comp = np.zeros((n_items, n_items))
+                for v2 in range(self._n_views):
+                    if v1 != v2:
+                        l_comp = l_comp + U_mats[v2] @ U_mats[v2].T
+                l_comp = (l_comp + l_comp.T) / 2
+                # Adding the symmetrized graph laplacian for view v1
+                l_mat = L_mats[v1] + self.v_lambda * l_comp
+                U_mats[v1], d_mat, _ = sp.sparse.linalg.svds(l_mat,
+                                                             k=self.n_clusters)
+                obj_vals[0, it] = np.sum(d_mat)
+
+            # Update U and the objective function value for view 1
+            l_comp = np.zeros((n_items, n_items))
+            for vi in range(self._n_views):
+                if vi != 0:
+                    l_comp = l_comp + U_mats[vi] @ U_mats[vi].T
+            l_comp = (l_comp + l_comp.T) / 2
+            l_mat = L_mats[0] + self.v_lambda * l_comp
+            U_mats[0], d_mat, _ = sp.sparse.linalg.svds(l_mat,
+                                                        k=self.n_clusters)
+            obj_vals[0, it] = np.sum(d_mat)
+            check_u_mats.append(U_mats[0])
+        self._objective = obj_vals
+
+        # Create final spectral embedding to cluster
+        V_mat = np.hstack(U_mats)
+        norm_v = np.sqrt(np.diag(V_mat @ V_mat.T))
+        norm_v[norm_v == 0] = 1
+        self._embedding = np.linalg.inv(np.diag(norm_v)) @ V_mat
+
+        # Perform kmeans clustering with embedding
+        kmeans = KMeans(n_clusters=self.n_clusters, n_init=self.n_init,
+                        random_state=self.random_state)
+        predictions = kmeans.fit_predict(V_mat)
+        return predictions
```

Comment:
Does this method not have a way to do .fit()? If not, can you add something, maybe to the Notes section, saying it doesn't implement fit(), only fit_transform together? See https://github.com/neurodata/mvlearn/blob/master/mvlearn/embed/mvmds.py for an example. Also, can you add this same thing to MultiviewSpectralClustering for consistency?

Let me know what you think about this.

Reply:
Ok, I decided follow the convention used by sklearn's spectral clustering algorithm which has both fit and fit_predict. The only functional difference between the two is that one returns the labels at the end and the other doesn't.